File: bubble-sort.py
tl = [1,4,5,6,3,8,9,7,2]

def bubble_sort(t):
  temp_s = 0
  temp_l = 0
  i = 0
  while i < len(tl) - 1:
    if tl[i] > tl[i+1]:
      temp_l = tl[i]
      tl[i] = tl[i+1]
      tl[i+1] = temp_l
      i -= 1
    else:
      i += 1
  return tl

bubble_sort(tl)

File: BubbleSort.java
import java.util.Arrays;

public class BubbleSort {
	public static void main(String[] args) {
		Integer[] a = { 2, 4, 1, 3, 9, 7 };
		System.out.println("Before: " + Arrays.toString(a));
		sort(a);
		System.out.println("After: " + Arrays.toString(a));
	}

	static <E extends Comparable<? super E>> void sort(E[] A) {
		for (int i = 0; i < A.length - 1; i++) // Bubble up i’th record
			for (int j = A.length - 1; j > i; j--)
				if ((A[j].compareTo(A[j - 1]) < 0)) {
					Util.swap(A, j, j - 1);
				}
	}
}
